dig.go

```diff
@@ -552,6 +594,10 @@ type node struct {
 	// Location where this function was defined.
 	location *digreflect.Func
 
+	// id is the address of memory location for the constructor. It is used to
```

Comment:
nit: We can remove the "the address ..." note and just have something similar to:
```
id uniquely identifies the constructor that produces a node.
```